pkg/features/features.go

```diff
@@ -137,7 +138,7 @@ var AllFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	NodeAgentAuthorizer:                      {Default: true, PreRelease: featuregate.Beta},
 	CredentialsRotationWithoutWorkersRollout: {Default: true, PreRelease: featuregate.Beta},
 	InPlaceNodeUpdates:                       {Default: false, PreRelease: featuregate.Alpha},
-	RemoveAPIServerProxyLegacyPort:           {Default: true, PreRelease: featuregate.Beta},
+	RemoveAPIServerProxyLegacyPort:           {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```

Comment:
Does it make sense to also clean up the feature gate branches as now the feature gate cannot be disabled anymore? See example in https://github.com/gardener/gardener/pull/11107/files#diff-51d6dcfca3b6142beb72780e52d0a554bff707dc2aeebe35df9edc6b05aa297fL184-R183.